### DISCUSSION TITLE
How to have expo-router `app` folder in `src/app`

### DISCUSSION THREAD

**Discussion by Akronae** [expo/router#374] [Q&A] [answered]
I want my routes (`app/**`) to be in a `src/` folders along with other directories, it would look something like that:
```
|- node_modules/
|- src/
|-- app/
|-- utils/
|-- index.js
|- package.json
|- .....
```

How can I achieve that with `expo-router`, setting `"main": "./src/index.js"` in `package.json` does not work.

**Comment by firasrg** [comment]
Issue already solved, check this [link](https://github.com/expo/router/issues/41#issuecomment-1406279131)

There is also [another discussion](https://github.com/expo/router/discussions/259) about this topic

**Reply by Akronae** [reply]
Oh wow I thought there would be a proper option haha, thanks! `expo-router` is indeed very opiniated

**Comment by EvanBacon** [comment, marked as answer]
[Expo Router v2](https://blog.expo.dev/introducing-expo-router-v2-3850fd5c3ca1) supports this https://docs.expo.dev/router/reference/src-directory/
